src/lib.rs

```diff
@@ -74,6 +74,16 @@ impl PrivateKey {
             key::EitherEd25519SecretKey::Extended(ref secret) => secret.to_bech32_str(),
         }
     }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<PrivateKey, JsValue> {
```

Comment:
I would go explicit on this and have 2 different functions form extended and non extended. So error handling can be done appropriately

Reply:
Okay. Just added a commit to split this